lib/qless/middleware/retry_exceptions.rb

```diff
@@ -48,19 +48,18 @@ def on_retry_callback
         @on_retry_callback ||= DEFAULT_ON_RETRY_CALLBACK
       end
 
-      def exponential(base, options = {})
+      # If`factor` is omitted it is set to `delay_s` to reproduce legacy
+      # behavior.
+      def exponential(delay_s, options={})
+        factor = options.fetch(:factor, delay_s)
         fuzz_factor = options.fetch(:fuzz_factor, 0)
 
-        lambda do |num, _error|
-          unfuzzed = base**num
+        lambda do |*unused_args|
+          unfuzzed = delay_s
+          delay_s *= factor
```

Comment:
Mutating `delay_s` seems problematic here because it means that the value continuously increases over the life time of the process, even as we run into different jobs that are failing and retrying with this exponential algorithm.  I think we need to use the `attempt_num` argument that is passed to the lambda instead.